src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4158,20 +4159,23 @@ private static BoundExpression SkipReferenceConversions(BoundExpression possibly
             BoundExpression leftOperand = node.LeftOperand;
             BoundExpression rightOperand = node.RightOperand;
 
-            TypeWithState leftResult = VisitRvalueWithState(leftOperand);
-            TypeWithState rightResult;
-
+            // PROTOTYPE(improved-definite-assignment):
+            // Definite assignment has another special case for when the LHS is a non-null constant (a literal string for example).
+            // Should we replicate that here?
             if (IsConstantNull(leftOperand))
             {
-                rightResult = VisitRvalueWithState(rightOperand);
+                VisitRvalue(leftOperand);
```

Comment:
I don't understand why we're visiting this. It's constant null, there's nothing to learn?

Reply:
Since we still need to record a result for the node, it seems like the simplest thing to do.

Reply:
Just use `SetResultX` instead.

---
In reply to: [620670774](https://github.com/dotnet/roslyn/pull/52646#discussion_r620670774) [](ancestors = 620670774)

Reply:
Done.

Reply:
Actually fixed it now.

Reply:
This change appears to break the following tests

- CondAccess_NullCoalescing_10
- NullCoalescingOperator_18

Basically it appears that when we use a `SetResultType` or similar approach with an expression like `(C?)null`, we set a result for the conversion node, but not for the operand node. I'll change it back to VisitRvalue because it seems like a straightforward way to traverse any child nodes of the constant expression and make sure we set results for all of them.